From 6614a7529d44c48fd5e8980eacab08c92cf1622b Mon Sep 17 00:00:00 2001
From: Venkata Jagadeesh Garaga <vgaraga@codeaurora.org>
Date: Thu, 27 Jun 2019 10:44:00 +0530
Subject: [PATCH 10/37] GAP: Reset bondingInitiatedLocally flag(1/3)

When Pairing initiated from DUT, bondingIinitiatedLocally
flag sets from adapterservice. If user disable profile
options from UI and tries to connect from remote,DUT
rejects the incoming connection, when dut rejects
connection some remotes may initiates pairing again.

Profile options should be in disable state after incoming
pairing successful as user disabled profile options, but
profile options getting enabled as bondingInitiatedLocally
not reset.

Hence Reset bondingInitiatedLocally flag after connection

 Conflicts:
	packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java

Change-Id: I612b0718fcc4f6aaeded112fee4cb51ad7f25772
CRs-Fixed: 2457374
Signed-off-by: Volodymyr Zhdanov <wight554@gmail.com>
---
 core/java/android/bluetooth/BluetoothDevice.java | 16 ++++++++++++++++
 .../bluetooth/CachedBluetoothDevice.java         |  9 +++++++--
 2 files changed, 23 insertions(+), 2 deletions(-)

diff --git a/core/java/android/bluetooth/BluetoothDevice.java b/core/java/android/bluetooth/BluetoothDevice.java
index 8b60a0d925c..e56a6b8bad0 100644
--- a/core/java/android/bluetooth/BluetoothDevice.java
+++ b/core/java/android/bluetooth/BluetoothDevice.java
@@ -1303,6 +1303,22 @@ public final class BluetoothDevice implements Parcelable {
         return false;
     }
 
+    /** @hide */
+    @UnsupportedAppUsage
+    public void setBondingInitiatedLocally(boolean localInitiated) {
+        final IBluetooth service = sService;
+        if (service == null) {
+            Log.w(TAG, "BT not enabled, setBondingInitiatedLocally failed");
+            return;
+        }
+        try {
+            service.setBondingInitiatedLocally(this, localInitiated);
+        } catch (RemoteException e) {
+            Log.e(TAG, "", e);
+        }
+        return;
+    }
+
     /**
      * Set the Out Of Band data for a remote device to be used later
      * in the pairing mechanism. Users can obtain this data through other
diff --git a/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java b/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java
index 04b84ab984c..2d4828dbff7 100644
--- a/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java
+++ b/packages/SettingsLib/src/com/android/settingslib/bluetooth/CachedBluetoothDevice.java
@@ -737,12 +737,17 @@ public class CachedBluetoothDevice implements Comparable<CachedBluetoothDevice>
         refresh();
 
         if (bondState == BluetoothDevice.BOND_BONDED) {
+            boolean mIsBondingInitiatedLocally = mDevice.isBondingInitiatedLocally();
+            Log.w(TAG, "mIsBondingInitiatedLocally" + mIsBondingInitiatedLocally);
+            if (mIsBondingInitiatedLocally) {
+                mDevice.setBondingInitiatedLocally(false);
+            }
             if (mIsTwsConnectEnabled) {
                 Log.d(TAG, "Initiating connection to" + mDevice);
-                if (mDevice.isBondingInitiatedLocally() || mDevice.isTwsPlusDevice()) {
+                if (mIsBondingInitiatedLocally || mDevice.isTwsPlusDevice()) {
                     connect();
                 }
-            } else if (mDevice.isBondingInitiatedLocally()) {
+            } else if (mIsBondingInitiatedLocally) {
                 connect();
             }
         }
-- 
2.28.0

